### PR TITLE
feat: interactive prompt escape sequence support

### PR DESCRIPTION
**Pull request by JanDeDobbeleer**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md]
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)

### Description

- refactor(text): escape segment text
- feat: interactive prompt sequences

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

**Comment by JanDeDobbeleer** [comment]
@lnu can you find some time to validate fish on your end? It somehow always behaved differently when dealing with this before.

@lewis-yeung look at this, the 4th iteration which could be successful and a lot simpler

**Comment by lewis-yeung** [comment]
@JanDeDobbeleer Found a bug in fish.

bug-in-fish

A tooltip with a hyperlink doesn't display correctly (extra backslash). This doesn't happen if we remove the following:

https://github.com/JanDeDobbeleer/oh-my-posh/blob/8253732f184d1ed9c12c353bc94df9b1d9e9c0e6/src/color/ansi.go#L125-L127

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung that's what's so strange about my terminal, it doesn't happen. We need to find a fix for the hyperlink without resorting to a full replace like before.

**Comment by JanDeDobbeleer** [comment]
@lewis-yeung I did it differently, can you try again?

**Comment by lewis-yeung** [comment]
@JanDeDobbeleer Validated. It works. 🎉

File: .vscode/launch.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
         "prompt",
         "print",
         "primary",
-        "--config=${workspaceRoot}/themes/default.omp.json",
+        "--config=~/.posh.omp.json",
```
Comment (lewis-yeung) [inline .vscode/launch.json]: Just curious, why do we need this change?

Comment (JanDeDobbeleer) [reply .vscode/launch.json]: We don't. Will remove.

**Pull request merged**